sdk/cognitivelanguage/ai-language-text/test/public/utils/customTestHelpter.ts

```diff
@@ -0,0 +1,171 @@
+// Licensed under the MIT license.
+
+import {
+  TextAnalysisAuthoringDeployProjectParameters,
+  TextAnalysisAuthoringTrainBodyParam,
+  getLongRunningPoller,
+  CreateProjectOptions,
+  TextAuthoringClient,
+  ExportedCustomSingleLabelClassificationProjectAssets,
+  ExportedCustomMultiLabelClassificationProjectAssets,
+  ExportedCustomEntityRecognitionProjectAssets,
+} from "@azure/ai-language-textauthoring";
+import { BlobServiceClient, ContainerClient } from "@azure/storage-blob";
+import { DefaultAzureCredential } from "@azure/identity";
+import path from "path";
+import decompress from "decompress";
+import {
+  customEntityAssets,
+  customMultiLabelAssets,
+  customSingleLabelAssets,
+} from "../customTestsAssets";
+import { assertEnvironmentVariable } from "@azure-tools/test-recorder";
+import { logger } from "../../../src/logger";
+import { readdir, rm } from "fs/promises";
+
+const pathName = path.join(
+  __dirname.split("ai-language-text")[0],
+  "ai-language-text",
+  "test",
+  "public",
+  "utils"
+);
+const storageInputContainerName = "documents";
+const language = "en-us";
+const trainingConfigVersion = "2022-05-01";
+const modelLabel = "projectModel";
+const trainParam: TextAnalysisAuthoringTrainBodyParam = {
+  body: {
+    modelLabel,
+    trainingConfigVersion,
+    evaluationOptions: {
+      kind: "percentage",
+      testingSplitPercentage: 30,
+      trainingSplitPercentage: 70,
+    },
+  },
+};
+const deployParam: TextAnalysisAuthoringDeployProjectParameters = {
+  body: {
+    trainedModelLabel: modelLabel,
+  },
+};
+
+async function createStorageBlob(): Promise<ContainerClient> {
+  /* Return a container client */
+  const defaultAzureCredential = new DefaultAzureCredential();
+  // Get documents containers
+  const blobServiceClient = new BlobServiceClient(
+    assertEnvironmentVariable("STORAGE_ENDPOINT"),
+    defaultAzureCredential
+  );
+  return blobServiceClient.getContainerClient(storageInputContainerName);
+}
+
+async function uploadDocumentsToStorage(dirName: string): Promise<void> {
+  const fileNames = await readdir(dirName);
+  const client = await createStorageBlob();
+  for (const file of fileNames) {
+    if (file.endsWith(".txt")) {
+      const blob = client.getBlockBlobClient(file);
+      const filePath = path.join(dirName, file);
+      await blob.uploadFile(filePath);
+    }
+  }
+  rm(dirName, { recursive: true, force: true });
+}
+
+async function getAssetsforProject(
+  projectKind:
+    | "CustomEntityRecognition"
+    | "CustomSingleLabelClassification"
+    | "CustomMultiLabelClassification"
+): Promise<{
+  assets:
+    | ExportedCustomSingleLabelClassificationProjectAssets
+    | ExportedCustomMultiLabelClassificationProjectAssets
+    | ExportedCustomEntityRecognitionProjectAssets;
+  dirName: string;
+}> {
+  switch (projectKind) {
+    case "CustomSingleLabelClassification":
+      return { assets: customSingleLabelAssets, dirName: path.join(pathName, "WebOfScience") };
```

Comment:
Should we move the path computation bit to happen outside the function so we don't have to repeat in every case?
```suggestion
      return { assets: customSingleLabelAssets, dirName: path.join(pathName, "WebOfScience") };
```

Reply:
Since each project has a different path, I'm putting it inside the switch statement

Reply:
Can we return just the name of the file since this is the only piece that changes between the cases and construct the path outside the function?

Reply:
That makes sense, I'll return the file name and do the path computation after the function returns. Thank you for the feedback!